sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -1417,8 +1428,69 @@ public void executeBundles(ProcessContext context) {
     }
   }
 
+  /** The type Patch resources. */
+  public static class PatchResources
+      extends PTransform<PCollection<FhirPatchParameter>, Write.Result> {
+    @Override
+    public FhirIO.Write.Result expand(PCollection<FhirPatchParameter> input) {
+      PCollectionTuple bodies =
+          input.apply(
+              ParDo.of(new PatchResourcesFn())
+                  .withOutputTags(Write.SUCCESSFUL_BODY, TupleTagList.of(Write.FAILED_BODY)));
+      bodies.get(Write.SUCCESSFUL_BODY).setCoder(StringUtf8Coder.of());
+      bodies.get(Write.FAILED_BODY).setCoder(HealthcareIOErrorCoder.of(StringUtf8Coder.of()));
+      return Write.Result.in(input.getPipeline(), bodies);
+    }
+
+    /** The type Write Fhir fn. */
+    static class PatchResourcesFn extends DoFn<FhirPatchParameter, String> {
+
+      private static final Counter PATCH_RESOURCES_ERRORS =
+          Metrics.counter(
+              PatchResourcesFn.class, BASE_METRIC_PREFIX + "patch_resources_error_count");
+      private static final Counter PATCH_RESOURCES_SUCCESS =
+          Metrics.counter(
+              PatchResourcesFn.class, BASE_METRIC_PREFIX + "patch_resources_success_count");
+      private static final Distribution PATCH_RESOURCES_LATENCY_MS =
+          Metrics.distribution(
+              PatchResourcesFn.class, BASE_METRIC_PREFIX + "patch_resources_latency_ms");
+
+      private transient HealthcareApiClient client;
+      private final ObjectMapper mapper = new ObjectMapper();
+
+      /**
+       * Initialize healthcare client.
+       *
+       * @throws IOException If the Healthcare client cannot be created.
+       */
+      @Setup
+      public void initClient() throws IOException {
+        this.client = new HttpHealthcareApiClient();
+      }
+
+      @ProcessElement
+      public void patchResources(ProcessContext context) {
+        FhirPatchParameter patchParameter = context.element();
```

Comment:
This looks fine. I wonder if it helps to group them into batches, and use a thread pool/async calls to perform multiple requests at the same time? It may improve utilization and throughput of an individual worker.

Reply:
Done, with a small numShards to help not exceed quota.